### PR TITLE
feat: data provider bsc token & BABT token holder group

### PR DESCRIPTION
**Pull request by mme022**
As mentioned in the comments of https://github.com/sismo-core/sismo-hub/issues/1483 here is a new data provider for BSC tokens and NFTs, powered by Ankr.
As their free version has a rate limit it takes a while to generate the group of around 700k BABT holders, but I couldn't find any other free provider that offers the information.

### CODE REVIEW

File: group-generators/helpers/data-providers/bsc-token-provider/index.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        pageToken: "",
+      },
+    };
+
+    do {
+      const data: BscTokenQueryResponse = await this.getHolders(
+        tokenRequestParams
+      );
+
+      console.log(data);
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/bsc-token-provider/index.ts:38]: There are some console.log in the code, it would be nice to remove them :)

**Review by MartinGbz — COMMENTED** [review]
left some comments

File: group-generators/helpers/data-providers/bsc-token-provider/index.ts (hunk 1 above)
Comment (mme022) [reply group-generators/helpers/data-providers/bsc-token-provider/index.ts:38]: omg.... so sorry. That mistakes shouldn't have been necessary. 😆

**Comment by MartinGbz** [comment]
> As mentioned in the comments of #1483 here is a new data provider for BSC tokens and NFTs, powered by Ankr. As their free version has a rate limit it takes a while to generate the group of around 700k BABT holders, but I couldn't find any other free provider that offers the information.

Yes I understand, maybe we can use other alternatives like [bitquery](https://bitquery.io/pricing) for example (I saw that we can fetch 10 free call per minute, maybe it can do the job?)

But yes it will be surely complicated to find a way to do this without paying a bit

**Comment by mme022** [comment]
I found out the one from Ankr can do up to 50req/min with 1000 addresses per request, that's the fastest I found. But I can have another deeper look.

**Comment by mme022** [comment]
I checked and another provider could be Moralis, but they expect to use their SDK and require an API key. Do you want to add moralis as depency to the hub and store and API key in a .env ? Otherwise I think 50.000 addresses/min with Ankr is the best I could find. :)

**Comment by MartinGbz** [comment]
Hey @mme022 I tested the code with a premium version of ankr but the problem still the same.
We can't paralellise multiple requests (for example with a promise.all(), because for each request, it gave us in response a "nextPage" field in order to fetch the next results). So we can't fetch all in once, that's why it takes ~15min to fetch all the data and that's unfortunately too long...

**Comment by mme022** [comment]
Wow still 15 mins with the timeout removed? That's sad to hear.

**Comment by MartinGbz** [comment]
> Wow still 15 mins with the timeout removed? That's sad to hear.

May bad! totally forgot this set timeout (was thinking the API was very slow, I'm dumb) 😅😅
I'll merge this in the afternoon! 👍

**Comment by mme022** [comment]
> > Wow still 15 mins with the timeout removed? That's sad to hear.
> 
> May bad! totally forgot this set timeout (was thinking the API was very slow, I'm dumb) 😅😅 I'll merge this in the afternoon! 👍

Ah I could've commented that here as well. ;) Nice! Great that it works then!

File: group-generators/helpers/data-providers/ankr/index.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    });
+
+    return res;
+  }
+
+  private checkArgsValidity(network: string, address: string) {
+    const regex =/^0x[0-9a-fA-F]{40}$/;
+    if(!regex.test(address)){
+      throw new Error(
+        `This is not a valid contract address`
```
Comment (yum0e) [inline group-generators/helpers/data-providers/ankr/index.ts:149]: can we have more information about the address that is not valid in the error message? Same thing for the network

File: group-generators/helpers/data-providers/ankr/index.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    do {
+      const data: AnkrTokenQueryResponse = await this.getHolders(
+        tokenRequestParams
+      );
+
+      // check if data is undefined
+      if(!data || !data.result) {
+        throw new Error(
+          `The fetched data is undefined
```
Comment (yum0e) [inline group-generators/helpers/data-providers/ankr/index.ts:51]: Can it be formatted nicely? Strange line format when triggering the error

**Review by yum0e — APPROVED** [review]

**Pull request merged**